tests/schema/tests.py

```diff
@@ -3095,6 +3095,30 @@ def test_add_field_default_dropped(self):
             if connection.features.can_introspect_default:
                 self.assertIsNone(field.default)
 
+    def test_add_field_default_nullable(self):
+        # Create the table
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        # Add new nullable CharField with a default
+        new_field = CharField(max_length=15, blank=True, null=True, default='surname')
+        new_field.set_attributes_from_name("surname")
+        with connection.schema_editor() as editor:
+            editor.add_field(Author, new_field)
+        Author.objects.create(name='Anonymous1')
+        # Ensure field is still nullable
+        with connection.cursor() as cursor:
+            cursor.execute("SELECT surname FROM schema_author;")
+            item = cursor.fetchall()[0]
+            self.assertIsNone(item[0])
+            field = next(
+                f for f in connection.introspection.get_table_description(cursor, "schema_author")
+                if f.name == "surname"
+            )
+            self.assertTrue(field.null_ok in (1, True))
```

Comment:
Ah I think you could use `assertTrue` without issues as both `1` and `True` are truthy.

```suggestion
            self.assertTrue(field.null_ok)
```